gk/fib.c

```diff
@@ -1487,33 +1464,74 @@ add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
 	 * accidentally create a security hole.
 	 */
 	rte_spinlock_lock_tm(&gk_conf->lpm_tbl.lock);
-	ret = check_prefix_locked(&prefix_info, action, gk_conf);
+	ret = check_prefix_locked(prefix_info, action, gk_conf);
 	if (ret < 0) {
 		rte_spinlock_unlock_tm(&gk_conf->lpm_tbl.lock);
 		return -1;
 	}
 
 	ret = add_fib_entry_locked(
-		&prefix_info, gt_para, gw_para, action, gk_conf);
+		prefix_info, gt_addr, gw_addr, action, gk_conf);
 	rte_spinlock_unlock_tm(&gk_conf->lpm_tbl.lock);
 
 	return ret;
 }
 
 int
-del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf)
+add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
+	enum gk_fib_action action, struct gk_config *gk_conf)
 {
 	int ret;
 	struct ip_prefix prefix_info;
+	struct ipaddr gt_addr, gw_addr;
+	struct ipaddr *gt_para = NULL, *gw_para = NULL;
 
-	prefix_info.str = ip_prefix;
-	prefix_info.len = parse_ip_prefix(ip_prefix, &prefix_info.addr);
+	prefix_info.str = prefix;
+	prefix_info.len = parse_ip_prefix(prefix, &prefix_info.addr);
 	if (prefix_info.len < 0)
 		return -1;
```

Comment:
If you move the two lines `prefix_info.str = prefix;` and `prefix_info.len = parse_ip_prefix(prefix, &prefix_info.addr);` just before the final return of `add_fib_entry()`, you can drop the whole statement `if (prefix_info.len < 0)` as you did in `del_fib_entry()`.